tests/resources/test_resources_communities.py

```diff
@@ -462,16 +461,17 @@ def test_accept_public_record_in_restricted_community(
     assert response.json["processed"]
     assert len(response.json["processed"]) == 1
     request_id = response.json["processed"][0]["request_id"]
-    client = uploader.logout(client)
-    client = community_owner.login(client)
 
-    # The error handlers for this action are defined in invenio-app-rdm, therefore we catch the exception here
-    with pytest.raises(InvalidAccessRestrictions):
-        client.post(
-            f"/requests/{request_id}/actions/accept",
-            headers=headers,
-            json={},
-        )
+    response = client.post(
+        f"/requests/{request_id}/actions/accept",
+        headers=headers,
+        json={},
+    )
+    assert response.status_code == 400
+    assert (
+        response.json["message"]
+        == "A public record cannot be included in a restricted community."
+    )
```

Comment:
Major: Not sure if this is a leftover, but the behaviour hasn't changed and tests fail with this change. I will revert it and we can check this later @anikachurilova

Reply:
Discussed IRL, there are changes needed due to the new permission